subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -2021,12 +2042,14 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		  "[-b, --band] 0: any band (2:2.4GHz, 5:5GHz, 6:6GHz]\n"
 		  "[-p, --psk]: Passphrase (valid only for secure SSIDs)\n"
 		  "[-k, --key-mgmt]: Key Management type (valid only for secure SSIDs)\n"
-		  "0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:EAP, 6:WEP,"
-		  " 7: WPA-PSK, 8: WPA-Auto-Personal\n"
+		  "0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:WEP, 6: WPA-PSK\n"
+		  "7: WPA-Auto-Personal, 8: EAP-TLS\n"
 		  "[-w, --ieee-80211w]: MFP (optional: needs security type to be specified)\n"
 		  ": 0:Disable, 1:Optional, 2:Required.\n"
 		  "[-m, --bssid]: MAC address of the AP (BSSID).\n"
 		  "[-t, --timeout]: Timeout for the connection attempt (in seconds).\n"
+		  "[-a, --anon-id]: Anonymous identity for enterprise mode.\n"
+		  "[-K, --key-passwd]: Private key passwd for enterprise mode.\n"
```

Comment:
How to add certs to wifi at build time? I think you can call the API to set the enterprise creds in your driver init phase.

Reply:
maybe this can be done in the future as enhancement?

Reply:
As discussed last in the Wi-Fi telco, you can put the raw cert in a file and can include it in a C array. See below links
https://github.com/nrfconnect/sdk-nrf/tree/v2.6.1/samples/net/aws_iot/certs
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/net/https_client/src/main.c#L46

> maybe this can be done in the future as enhancement?
I think you can call the API to set the enterprise creds in your driver init phase.

It would be good to get this in this PR itself, it shouldn't be too hard, this way anyone an test those APIs using shell rather than writing code to invoke the API.

Reply:
@krish2718 I can put the cert/key as C array in header files. but where should I put the header file? and what is the default value of the cert/key, just the one we currently use?

Reply:
> but where should I put the header file

It's a bit tricky, but I would just put them in `subsys/net/l2/wifi/test_certs` and include those in `wifi_shell.c` if Enterprise is enabled and use in `set_enterprise_creds`. @jukkar WDYT? The ideas is to use `shell` as primary test method.

> and what is the default value of the cert/key, just the one we currently use?

Sure, or you generate a new test ones.

Reply:
Hi @krish2718  @jukkar , updated as your comment, please review again

Reply:
Thanks, this is looking good, but it would be good to commit certs directly and then have a script or some way to generate the headers, this makes it easier to update these certs. WDYT?

I used this script:

```
 cat ~/crt_to_header.py 
#!/usr/bin/env python3
import sys

cer_file = sys.argv[1]
header_file = sys.argv[2]
macro_name = "CA_CERTIFICATE"

key = open(cer_file, "r")
c_file = open(header_file, "w")
c_file.write("#define " + macro_name + " \\\n")
for line in key:
    line = line.replace("\n", "")
    c_line = "\"" + line + "\\n\" \\\n"
    c_file.write(c_line)
print('Certificate converted to C header file in:', header_file)
key.close()
c_file.close()
```

This is based on https://academy.nordicsemi.com/courses/cellular-iot-fundamentals/lessons/lesson-4-cellular-fundamentals/topic/lesson-4-exercise-2/ (#3.2). No changes to source code, just replace cert and then run the script.

Reply:
Hi @krish2718 , if directly using certs file to replace the header files, then the build can not pass:
![image](https://github.com/user-attachments/assets/bf80f944-8d6c-43c3-85af-b2c5fde1e990)
as we need to use the C array in header files. why not let the user to use their way to generate the header files? 
we use this way:

Convert CA format form PEM to DER.
openssl x509 -outform der -in cas.pem -out ca.der
openssl x509 -outform der -in wifiuser.pem -out client.der

Convert RSA Key format to DER.
openssl rsa -in wifiuser.key -outform DER -out client_key.der

Convert certs to header file:
Convert .der to .h: xxd -i ca.der > ca.h

Reply:
I am suggesting to keep both the original cert file and the generated header, not just the cert file. And also commit the script, so, user can just update cert file run and generate header.

Reply:
can I add the pem file and header file, and a readme.txt to tell user how to convert pem to header file?
Convert CA format form PEM to DER.
openssl x509 -outform der -in cas.pem -out ca.der
openssl x509 -outform der -in wifiuser.pem -out client.der

Convert RSA Key format to DER.
openssl rsa -in wifiuser.key -outform DER -out client_key.der

Convert certs to header file:
Convert .der to .h: xxd -i ca.der > ca.h

Reply:
Hi @krish2718  please review again